### PR TITLE
Deprecate summary and content attributes, and fix description, summary, content text formats in the RSS feeds

### LINKED ISSUE #1030: Problems with `<itunes:summary\>`, `<content:encoded\>`, `<description\>` and apple podcasts preview pages
Apple renders the podcast preview pages using a different set of rules for the description text/html than in the apps.

The native apps now seem to only look at `<description/>`, and allow some html including `<p>` tags for new lines.
https://help.apple.com/itc/podcasts_connect/#/itcb54353390

It says just these tags, but it also seems to respect `<br>` tags:
```
description is text containing one or more sentences describing your episode to potential listeners. You can specify up to 4000 bytes. You can use rich text formatting and some HTML (<p>, <ol>, <ul>, <li>, <a>) if wrapped in the <CDATA> tag.
```

We also have fields in the episodes db table for `episodes.content` (`<content:encoded \>`) and `episodes.summary` (`<itunes:summary \>`), but I don't think the new feeder UI sets those.
However, we have old values in there, from when publish would set them and they still come in on rss import.

I also see in the ui, that there are hidden fields that will set episode `content` and `summary` to nil whenever an episode is saved.
This seems to indicate we want to move away from those, and it will cause any `<content:encoded />`  to stop being included once that is set to `nil`, but the rss builder will still set an `<itunes:summary/>` based on a sanitized `description`.

This further is an issue because the sanitization procedure replaces `<br\>` and `<p\>` tags but doesn't convert them to spaces or newlines, so it causes words to get concatenated that were previously separated by the whitespace form those tags.

So there are a few things we could do - 

1. get rid of displaying `<content:encoded>` and `<itunes:summary>` entirely - and just rely on `<description>`.

2. add back `<content:encoded>` but have it always identical to description, this will then take precedence over summary if that value remains 

3. if we keep summary, we need to add content encoded, and we need to fix the white space issue with the sanitizer

### PR DESCRIPTION
**Pull request by kookster**
fixes #1030 using option 3 from that ticket

### CODE REVIEW

File: app/models/concerns/import_utils.rb [hunk 1]
```diff
@@ -110,9 +112,7 @@ def remove_podcastchoices_link(str)
   end
 
   def sanitize_html(text)
-    return nil if text.blank?
-    sanitizer = Rails::Html::WhiteListSanitizer.new
-    sanitizer.sanitize(Loofah.fragment(text).scrub!(:prune).to_s).strip
+    sanitize_white_list(text)
```
Comment (kookster) [inline app/models/concerns/import_utils.rb:115]: might as well just use this so it is consistent

File: app/models/concerns/text_sanitizer.rb [hunk 2]
```diff
@@ -37,4 +42,20 @@ def sanitize_category(kw, max_length, strict)
...
+      .gsub(/<\/p>/i, "</p>\n")
+      .gsub(/<div>/i, "\n<div>")
+      .gsub(/<\/div>/i, "</div>\n")
+      .gsub(/<br>/i, "\n<br>")
+      .gsub(/<br\s*\/>/i, "\n<br>")
+  end
+
+  def clean_whitespace(text)
+    text.tr("\r", "\n")
+      .gsub(/\s*\n\s*/, "\n")
```
Comment (kookster) [inline app/models/concerns/text_sanitizer.rb:57]: the goal here is to end up with a single newline, even if there were multiple p or br tags stacked
arguably, we could support multiple newlines...that would be another way to go, but I don't see a good reason for it in the text versions of these descriptions

File: app/models/imports/episode_rss_import.rb [hunk 3]
```diff
@@ -110,7 +107,7 @@ def entry_description(entry)
   end
 
   def entry_description_attribute(entry)
-    [:content, :itunes_summary, :description, :title].find { |d| !entry[d].blank? }
+    [:content, :description, :itunes_summary, :itunes_subtitle, :title].find { |d| !entry[d].blank? }
```
Comment (kookster) [inline app/models/imports/episode_rss_import.rb:110]: content and description consistently have the full html version much more often than summary

File: app/views/podcasts/show.rss.builder [hunk 4]
```diff
@@ -155,7 +155,7 @@ xml.rss "xmlns:atom" => "http://www.w3.org/2005/Atom",
               url: ep.enclosure_url(@feed))
           end
 
-          xml.content(:encoded) { xml.cdata!(ep.content) } unless ep.content.blank?
+          xml.content(:encoded) { xml.cdata!(episode_description(ep)) }
```
Comment (cavis) [inline app/views/podcasts/show.rss.builder:158]: So now `<content:encoded>` will be identical to `<description>` - do we really need both?

File: app/models/concerns/text_sanitizer.rb [hunk 5]
```diff
@@ -37,4 +42,21 @@ def sanitize_category(kw, max_length, strict)
...
     end
   end
+
+  def add_newlines_to_tags(text)
+    text.gsub(/<p>/i, "\n<p>")
+      .gsub(/<\/p>/i, "</p>\n")
+      .gsub(/<div>/i, "\n<div>")
+      .gsub(/<\/div>/i, "</div>\n")
+      .gsub(/<br>/i, "\n<br>")
+      .gsub(/<br\s*\/>/i, "\n<br>")
```
Comment (cavis) [inline app/models/concerns/text_sanitizer.rb:52]: Should these check if the tags already have newlines?  Or do we scrub those out before calling this?

File: app/models/concerns/text_sanitizer.rb [hunk 6]
```diff
@@ -37,4 +42,21 @@ def sanitize_category(kw, max_length, strict)
...
+  end
+
+  def clean_whitespace(text)
+    text.gsub(/\R/, "\n") # if there is any kind of newline, make it \n
+      .gsub(/([ \t]*)\n([ \t]*)/, "\n") # get rid of whitespace on eother side of a newline
+      .gsub(/\n\n\n/, "\n\n") # no more than 2 newlines together
```
Comment (cavis) [inline app/models/concerns/text_sanitizer.rb:58]: Ah, maybe this covers my previous questions... even if we had too many newlines, this should remove some of them.

Are these rules from some real-world cases you've seen, or just from the test?

I also wonder if the last steps (trimming/combining spaces) should happen before the multiple-newline detection.  But this is probably fine.

**Review by cavis — APPROVED** [review]
Couple questions, but this looks good to me!

:+4:

File: app/models/concerns/text_sanitizer.rb (hunk 5 above)
Comment (kookster) [reply app/models/concerns/text_sanitizer.rb:52]: it dedupes the newlines anyway, so adding another is always fine.

File: app/models/concerns/text_sanitizer.rb (hunk 6 above)
Comment (kookster) [reply app/models/concerns/text_sanitizer.rb:58]: the `strip` could be before or after I guess, doesn't seem to matter much.
the prior regex gets rid of spaces near newlines the final squeeze is any place with multiple spaces, so again different, and could be any order

File: app/views/podcasts/show.rss.builder (hunk 4 above)
Comment (kookster) [reply app/views/podcasts/show.rss.builder:158]: I see multiple rss hosts that do provide both

File: app/models/concerns/text_sanitizer.rb (hunk 5 above)
Comment (cavis) [reply app/models/concerns/text_sanitizer.rb:52]: Makes good sense!

File: app/models/concerns/text_sanitizer.rb (hunk 6 above)
Comment (cavis) [reply app/models/concerns/text_sanitizer.rb:58]: +1

**Review by cavis — COMMENTED** [review]
:+onwards!:

**Pull request merged**